src/Jackalope/Transport/DoctrineDBAL/CachedClient.php

```diff
@@ -84,29 +84,27 @@ protected function workspaceExists($workspaceName)
     public function getNodeTypes($nodeTypes = array())
     {
         $cacheKey = 'nodetypes: '.serialize($nodeTypes);
-        $nodeTypes = $this->caches['meta']->fetch($cacheKey);
-        if (!$nodeTypes) {
-            $nodeTypes = parent::getNodeTypes($nodeTypes);
-            $this->caches['meta']->save($cacheKey, $nodeTypes);
+        $result = $this->caches['meta']->fetch($cacheKey);
+        if (!$result) {
+            $result = parent::getNodeTypes($nodeTypes);
+            $this->caches['meta']->save($cacheKey, $result);
         }
 
-        return $nodeTypes;
+        return $result;
     }
 
     /**
-     * Return the namespaces of the current session as a referenceable ArrayObject.
-     *
-     * @return \ArrayObject
+     * {@inheritDoc}
      */
-    protected function getNamespacesObject()
+    public function getNamespaces()
     {
-        if ($this->namespaces->count() === 0) {
+        if (!isset($this->namespaces)) {
```

Comment:
Does this effectively check if `namespaces` is null? If so I think `null !== $this->namespaces` might be clearer.